### ISSUE TITLE
buggy configure.py

### ISSUE THREAD

**Issue by maxpat78** [randombit/botan#3307]
C:\> py configure.py
or
C:> configure.py
--> error: Python not found etc.

Are you kidding us? What kind of checks is your script doing? My python 3.11 works fine for everything, except this!
In the past the configuration process was *really* an easy matter!

**Comment by maxpat78** [comment]
But `py -d configure.py` succeeds!
_Credo quia absurdum_

**Comment by reneme** [comment]
The problem is that the `configure.py` shebang reads `#!/usr/bin/env python` which has issues on systems that don't come with a `python` executable anymore. This is fixed on `master` but hasn't been backported to the Botan 2.x branch so far. Sorry for the inconvenience.

**Comment by randombit** [comment]
Can you be a bit more specific about what error you are seeing?

Also what is `py`?

**Comment by maxpat78** [comment]
`Python non trovato; Esegui senza argomenti per installare da Microsoft Store o disabilitare il collegamento da impostazioni > Gestisci app Alias di esecuzione.`
(Python not found; Invoke without arguments to install from Microsoft Store or to disable the link in Options > Manage app Execution alias) - a strange, never seen message.

py is a tool to invoke python, eventually selecting a specific version.

However, I've got the same error invoking `configure.py` or `python configure.py` - i.o.w. it is a script bug, not depending on my particular python setup.

**Comment by randombit** [comment]
This seems like a local installation issue?

**Comment by maxpat78** [comment]
No, it isn't.

In fact, removing the shebang `#!/usr/bin/env python` from the script, as suggested by reneme, fixes the problem.

I simply forgot to check the *first* script line and fix the issue myself...

**Comment by reneme** [comment]
Technically, we could backport #3046 to the `release-2` branch and sunset support for Python 2 for Botan 2.19.4 as a hotfix. Not sure, we'd want to do that, though, as it is somewhat a breaking change. But then again, Python 2 is gone for good for a while now.

So far I didn't find an alternative to changing the shebang apart from wrapping the python script with some platform specific magic that picks the correct interpreter.

**Comment by lieser** [comment]
Although the current recommendation is to be more specific (about using Python 2 or Python 3), the  `#!/usr/bin/env python` shebang is I think still the one that makes the most sense if the script still supports both Python versions.

And although there are some platforms there `#!/usr/bin/env python` does not get by default mapped to an installed Python 3 even if it is the only installed one, in general `py` does not have any problem with it. E.g. on my Windows, with only Python 3 installed, it works without a problem.

So although we may want to backport the dropping of Python 2 support, the current shebang itself is IMHO not a defect of the configure script, at least as long as we intentionally keep the Python 2 alive there.
Instead it is your local environment that does not properly handle this valid shebang.

You could try to be explicit about wanting to use python 3 by calling `py -3 configure.py`, or check your configuration (https://docs.python.org/3/using/windows.html#customization).

See also https://stackoverflow.com/a/19305076 and https://peps.python.org/pep-0394/#recommendation.

**Comment by maxpat78** [comment]
As I stated in my first comment, I have Python 3.11 (and it was the only Python installation).

However, you're right: `py -3 configure.py` also does the job; other legitimate calls (`configure.py`, `py configure.py`, `python configure.py`) give the error.

And now that I've just installed it, even `py -2 configure.py` works.
In other words, it seems that `py` alone fails, but with any option (i.e. `py -d configure.py` above) succeeds.

There is also a weird thing to say: issuing `python` or `python3` on my Win11 22H2 CMD prompt opens the Microsoft Store, suggesting me to install Python 3.10 - perhaps that is the real reason for the strange error message appeared 😵 - (but `py` correctly starts Python 3.11).
